### ISSUE TITLE
sql: aggregation hits memory error instead of spilling to disk

### ISSUE THREAD

**Issue by wzrdtales** [cockroachdb/cockroach#46113]
**Describe the problem**
When I throw a query at the DB like the following one:

```
select so, COUNT(some) as count, min(some) AS da from newtable GROUP BY so HAVING COUNT(some) > 3
```

it will crash with the following error:

```
pq: root: memory budget exceeded: 10240 bytes requested, 134215680 currently allocated, 134217728 bytes in budget
```

This was not the case in earlier versions of cockroach. I am using the same query on the same table as before (it is a really huge table), but now it will not even try to execute the query anymore. Last known version where this behavior was still normal is `v19.1.3`

**Environment:**
 - CockroachDB version 19.2.2

**Additional context**
What was the impact?

The DB is completely useless now... . Need to consider other technologies again since downgrading is not an option.

**Comment by yuzefovich** [comment]
This seems like #42485 - we are missing spilling to disk in the hash aggregator in the row-by-row execution engine. Could you please run `EXPLAIN` on this query to confirm this?

Also, since 19.1.3 version was working, that the new vectorized engine is to blame here, and the explain will tell us whether the vectorized engine is used.

As one remedy step, you can try increasing `--max-sql-memory` argument when starting up CockroachDB cluster. The default value is `128MB` and is quite low, starting from 20.1 version the default will be `.25` (meaning a quarter of all available RAM on the machine).

**Comment by wzrdtales** [comment]
```
       tree      |    field    |    description     
+----------------+-------------+-------------------+
                 | distributed | true               
                 | vectorized  | false              
  filter         |             |                    
   │             | filter      | count > 3          
   └── group     |             |                    
        │        | aggregate 0 | id          
        │        | aggregate 1 | count(thisid)  
        │        | aggregate 2 | min(someid)    
        │        | group by    | id          
        └── scan |             |                    
                 | table       | tablenew@primary       
                 | spans       | ALL            
```

**Comment by wzrdtales** [comment]
Verbose one

```
       tree      |    field    |    description    |               columns               | ordering  
+----------------+-------------+-------------------+-------------------------------------+----------+
                 | distributed | true              |                                     |           
                 | vectorized  | false             |                                     |           
  filter         |             |                   | (id, count, cid)             |           
   │             | filter      | count > 3         |                                     |           
   └── group     |             |                   | (id, count, cid)             |           
        │        | aggregate 0 | id         |                                     |           
        │        | aggregate 1 | count(thisid) |                                     |           
        │        | aggregate 2 | min(someid)   |                                     |           
        │        | group by    | id         |                                     |           
        └── scan |             |                   | (id, someid, thisid) |           
                 | table       | tablenew@primary      |                                     |           
                 | spans       | ALL               |                                     |    
```

**Comment by yuzefovich** [comment]
Could you please also provide `EXPLAIN (opt, verbose)` on the query (maybe even `EXPLAIN (opt, verbose, env)` if you're ok sharing with us the statistics on the tables involved in the query)?

**Comment by wzrdtales** [comment]
Unfortunately don't have the db here anymore, will need to reproduce it again.

**Comment by yuzefovich** [comment]
Closing as likely duplicate of #42485 (which will be addressed in 21.1).
